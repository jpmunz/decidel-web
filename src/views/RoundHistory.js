import React from "react";
import ACTION_TYPES from "src/actionTypes";
import wrench from "src/assets/wrench.svg";
import remove from "src/assets/remove.svg";
import "./RoundHistory.scss";

class RoundHistory extends React.Component {
  renderHistoryItem(item, i) {
    let actionText;
    let icon;

    if (item.action === ACTION_TYPES.ADDED) {
      actionText = "threw a wrench by adding";
      icon = wrench;
    } else if (item.action === ACTION_TYPES.REMOVED) {
      actionText = "crossed off";
      icon = remove;
    }

    return (
      <li className="round-item" key={i}>
        <img src={icon} alt="" /> {item.userName} {actionText} {item.title}
      </li>
    );
  }

  render() {
    return (
      <div className="RoundHistory">
        <ul>{this.props.history.map(this.renderHistoryItem.bind(this))}</ul>
      </div>
    );
  }
}

export default RoundHistory;
