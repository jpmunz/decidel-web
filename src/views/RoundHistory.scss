@import "mixins";

.RoundHistory {
  .round-item {
    &:not(:first-child) {
      margin-top: 10px;
    }

    img {
      width: 20px;
      margin-right: 10px;
    }
  }
}
