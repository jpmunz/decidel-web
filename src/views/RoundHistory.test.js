import React from "react";
import { render } from "@testing-library/react";
import RoundHistory from "./RoundHistory";

test("Basic rendering", () => {
  expect(
    render(
      <RoundHistory
        history={[
          { userName: "Bob", action: "ADDED", title: "option1" },
          { userName: "Bob", action: "REMOVED", title: "option2" }
        ]}
      />
    ).asFragment()
  ).toMatchSnapshot();
});
