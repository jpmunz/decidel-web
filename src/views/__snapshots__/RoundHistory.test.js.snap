// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic rendering 1`] = `
<DocumentFragment>
  <div
    class="RoundHistory"
  >
    <ul>
      <li
        class="round-item"
      >
        <img
          alt=""
          src="wrench.svg"
        />
         Bob threw a wrench by adding option1
      </li>
      <li
        class="round-item"
      >
        <img
          alt=""
          src="remove.svg"
        />
         Bob crossed off option2
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
